src/main/scala/com/gu/i18n/CountryGroup.scala

```diff
@@ -324,12 +327,11 @@ object CountryGroup {
 
   def byOptimisticCountryNameOrCode(str: String): Option[Country] = {
     val clean = str.replace(".", "")
-    val asCode = clean.toUpperCase
     val name = clean.toLowerCase
 
-    countryByName(name) orElse countryByCode(asCode) orElse (name match {
+    countryByName(name) orElse countryByCode(clean) orElse (name match {
       case _ if name endsWith "of ireland" => Some(Country.Ireland)
-      case _ if asCode == "GB" => Some(Country.UK)
+      case _ if clean == "GB" => Some(Country.UK)
```

Comment:
I would just use if else instead of a match that doesn't match anything but if you like it better this way fine